ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/ros2/RDXHeightMapVisualizer.java

```diff
@@ -68,21 +69,33 @@ public class RDXHeightMapVisualizer extends RDXVisualizer
 
    public RDXHeightMapVisualizer()
    {
-      super("Height Map");
+      this("Height Map");
+   }
+
+   public RDXHeightMapVisualizer(String title)
+   {
+      super(title);
 
       executorService = Executors.newFixedThreadPool(4);
    }
 
+   /** @deprecated Use RDXPerceptionVisualizerPanel which supports demand heartbeats.*/
    public void setupForHeightMapMessage(ROS2PublishSubscribeAPI ros2)
    {
       ros2.subscribeViaCallback(PerceptionAPI.HEIGHT_MAP_OUTPUT, this::acceptHeightMapMessage);
-      activeHeartbeat = new ROS2Heartbeat(ros2, PerceptionAPI.REQUEST_HEIGHT_MAP);
+      activeHeartbeat = new ROS2Heartbeat(ros2, PerceptionAPI.REQUEST_OUSTER_HEIGHT_MAP);
    }
 
    public void setupForImageMessage(ROS2Helper ros2)
    {
       ros2.subscribeViaCallback(PerceptionAPI.HEIGHT_MAP_CROPPED, this::acceptImageMessage);
-      activeHeartbeat = new ROS2Heartbeat(ros2, PerceptionAPI.REQUEST_HEIGHT_MAP);
+   }
+
+   /** @deprecated Use RDXPerceptionVisualizerPanel which supports demand heartbeats.*/
```

Comment:
@TomaszTB I think this is right?

Reply:
Yeah, seems right to me. I'd have to look at the context of this method to really make sure. Is it simply used to assign a heartbeat to this visualizer? If so, the comment is correct.

Reply:
Why are we deprecating stuff? And also what does demand heartbeats do? I ask because if using demand heartbeats is the thing we should be doing I would like to know what that is lol